src/Input/Silk.NET.Input.Desktop/Util.cs

```diff
@@ -0,0 +1,82 @@
+// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Linq;
+using Silk.NET.GLFW;
+using Silk.NET.Input.Common;
+using Ultz.Dispatcher.Unsafe;
+
+namespace Silk.NET.Input.Desktop
+{
+    internal static class Util
+    {
+        private static string[] _glfwKeys = Enum.GetNames(typeof(Keys));
+        public static object Do(Delegate @delegate)
+        {
+            return GlfwProvider.ThreadDispatcher.Invoke(@delegate);
+        }
+        
+        public static void Do(Action @delegate)
+        {
+            GlfwProvider.ThreadDispatcher.Invoke(@delegate);
+        }
+
+        public static T Do<T>(Func<T> @delegate)
+        {
+            return GlfwProvider.ThreadDispatcher.Invoke(@delegate);
+        }
+
+        public static unsafe T* Do<T>(Func<UnsafeDispatch<T>> @delegate)
+            where T:unmanaged
+        {
+            return GlfwProvider.ThreadDispatcher.Invoke(@delegate);
+        }
+
+        public static float ApplyDeadzone(Deadzone deadzone, float val)
```

Comment:
Personally, I feel this would be better suited to be put on the Deadzone struct itself, or at least in Input.Common somewhere since every backend would need it.

Reply:
Will do!